### ISSUE TITLE
indentation_linter thrown off by comment

### ISSUE THREAD

**Issue by AshesITR** [r-lib/lintr#1751]
Is the indentation lint here a false positive? AFAICT, the indentation should be 6 spaces here.

> Indentation should be 4 spaces but is 6 spaces.

Btw, if I try to inline this, I create `line_length_linter()` lint.

__Originally posted by @IndrajeetPatil in https://github.com/r-lib/lintr/pull/1748#discussion_r1002654036__

This should not lint:

```r
a <- # comment
  42L
```
